src/DataTransformer/ExternalUserActivationCodeInputDataTransformer.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace App\DataTransformer;
+
+use ApiPlatform\Core\DataTransformer\DataTransformerInterface;
+use App\Dto\UserActivationCodeInput;
+use App\Entity\Tenant\UserActivationCode;
+use App\Entity\User;
+use App\Exceptions\CodeGenerationException;
+use App\Repository\UserActivationCodeRepository;
+use App\Repository\UserRepository;
+use App\Service\ExternalUserService;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Component\Security\Core\Security;
+
+class ExternalUserActivationCodeInputDataTransformer implements DataTransformerInterface
+{
+    public function __construct(
+        private readonly Security $security,
+        private readonly ExternalUserService $externalUserService,
+        private readonly UserRepository $userRepository,
+        private readonly UserActivationCodeRepository $userActivationCodeRepository,
+    ) {}
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CodeGenerationException
+     *
+     * @var UserActivationCodeInput
+     */
+    public function transform($object, string $to, array $context = []): UserActivationCode
+    {
+        /** @var User $user */
+        $user = $this->security->getUser();
+
+        $roles = [];
+
+        // Only allow EXTERNAL_USER roles.
+        if (in_array('ROLE_EXTERNAL_USER_ADMIN', $object->roles)) {
+            $roles[] = 'ROLE_EXTERNAL_USER_ADMIN';
+        } else {
+            $roles[] = 'ROLE_EXTERNAL_USER';
+        }
+
+        $code = new UserActivationCode();
+        $code->setCode($this->externalUserService->generateExternalUserCode());
+        $code->setTenant($user->getActiveTenant());
+        // Expire: 2 days
+        $code->setCodeExpire((new \DateTime())->add(new \DateInterval('P2D')));
```

Comment:
Interval should be set as a class constant.

Nice to have: Interval can be set in `.env` instead

Reply:
Update: See comment for `ExternalUserService`

Reply:
Fixed